SCons/Tool/compilation_db.xml

```diff
@@ -45,30 +45,55 @@ See its __doc__ string for a discussion of the format.
     <builder name="CompilationDatabase">
         <summary>
             <para>
-                The &b-CompilationDatabase; builder writes a JSON formatted compilation
-                    database according to the
-                <ulink url="https://clang.llvm.org/docs/JSONCompilationDatabase.html">LLVM specification
-                </ulink> which is consumed by a number of clang tools, editors, and other tools.
-            </para>
-            <para>
-                If you don't specify any files, the builder will default to <filename>compile_commands.json</filename>.
+                &b-CompilationDatabase; is a special builder which
+                adds a target to create a JSON formatted
+                compilation database compatible with
+                <systemitem>clang</systemitem> tooling
+                (see the
+                <ulink url="https://clang.llvm.org/docs/JSONCompilationDatabase.html">LLVM specification</ulink>).
+                This database is suitable for consumption by various
+                tools and editors who can use it to obtain build and
+                dependency information which otherwise would be
+                internal to &SCons;.
+                The builder does not require any source files to be specified,
+                rather it arranges to emit information about all
+                of the C, C++ and assembler source/output pairs
+                identified in the build that are not excluded by the
+                optional filter &cv-link-COMPILATIONDB_PATH_FILTER;.
+                The target is subject to the usual &SCons; target
```

Comment:
I'd omit this sentence. I'm not sure it won't be confusing.

Reply:
I debated with myself - wrote something longer for the Note section earlier.  Since we've heard from one of the limited number of users so far of this feature that its absence caused confusion, I thought I'd leave a brief bit in the proposal.  We can leave it out if the user guide content makes this clear enough.

Reply:
I guess the problem is, that it's not strictly a builder in any normal sense.

Reply:
trying to make the best of explaining it...